sc-memory/sc-memory/sc_scs_helper.cpp

```diff
@@ -169,7 +169,26 @@ class StructGenerator
         {
           SetSCsGlobalIdtf(el.GetIdtf(), result);
         }
-        
+      }
+      else
+      {
+        ScType const & newType = el.GetType();
+        ScType const & oldType = m_ctx.GetElementType(result);
+        if (newType != oldType)
+        {
+          if (oldType.CanExtendTo(newType))
+          {
+            m_ctx.SetElementSubtype(result, *newType);
+          }
+          else
+          {
+            //Type NodeConst means that element have been called by system Id, that is why need to ignore this case as error
+            if (newType != ScType::NodeConst)
```

Comment:
Remove this if and comment

Reply:
if we remove this condition, the builder will not work